src/Spice86.Core/Emulator/Memory/Memory.cs

```diff
@@ -1,27 +1,259 @@
 ﻿namespace Spice86.Core.Emulator.Memory;
 
 using Spice86.Core.Emulator.Errors;
+using Spice86.Core.Emulator.VM;
 using Spice86.Core.Emulator.VM.Breakpoint;
 
 using System;
+using System.Buffers;
 using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
 
 /// <summary> Addressable memory of the machine. </summary>
 public class Memory {
+    private readonly uint _addressMask = 0x000FFFFFu;
+
+    /// <summary>
+    /// Starting physical address of video RAM.
+    /// </summary>
+    private const int VramAddress = 0xA000 << 4;
+
+    /// <summary>
+    /// The highest address which is mapped in the class VgaCard
+    /// </summary>
+    /// <remarks>
+    /// Video RAM mapping is technically up to 0xBFFF0 normally.
+    /// </remarks>
+    private const int VramUpperBound = 0xBFFF << 4;
+
+    /// <summary>
+    /// Size of conventional memory in bytes.
+    /// </summary>
+    public const uint ConvMemorySize = 1024 * 1024;
+
+    public bool EnableA20 { get; internal set; }
```

Comment:
This is actually an interesting property. 
 - If disabled, memory accesses should wrap when address > 1M
 - If enabled, memory addresses > 1M give access to a bit more ram (HMA, more info here: http://flint.cs.yale.edu/feng/cos/resources/BIOS/himem_A20.htm)